### PR TITLE
pre-commit: MD formatting

### PR DESCRIPTION
**Pull request by Borda**
add pre-commit hook for MarkDown formatting

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Updates to code quality tools, documentation improvements, and markdown formatting enhancements.

### 📊 Key Changes
- ⬆️ **Upgraded dependencies** for `pyupgrade`, `mdformat`, `yesqa`, and `flake8` to newer versions.
- 👌 **Enabled markdown formatting** tools by adding `mdformat` configuration and dependencies.
- ✅ **Quality Assurance addition** with `yesqa` tool enabled for unused `noqa` comments clean-up.
- ✍️ **Documentation** amended in `CONTRIBUTING.md`, `README.md`, and `wandb` README with minor readability improvements and updated usage instructions.
- 🧹 **Cleaned up commented-out sections** in various configuration files to improve clarity and maintainability.

### 🎯 Purpose & Impact
- 🚀 Users will benefit from **more robust code quality checks**, keeping the codebase modern and maintainable.
- ✨ Contributors will now be able to format **markdown files automatically**, ensuring consistent documentation style.
- 🧽 **Streamlined documentation** makes it easier for new users and contributors to understand and interact with the project.
- The ⚙️ **development pipeline** is leaner due to the clean-up of TODO items and old comments.

### CODE REVIEW

**Comment by glenn-jocher** [comment]
@Borda thanks for the PR! The markdown formatting all looks good except for the table changes. 

Is there a way to disable table updates? Or maybe I can manually revert the table changes just for this PR?

**Comment by Borda** [comment]
> The markdown formatting all looks good except for the table changes.

yes, I also prefer the short table...

> Is there a way to disable table updates? Or maybe I can manually revert the table changes just for this PR?

I do not think that manual fix would work as the bot would change it in each PR/commit

**Comment by Borda** [comment]
@glenn-jocher I do not think we can do anything about the tables, just accept it or reject whole MD formatting :]

File: README.md [hunk 1]
```diff
@@ -191,10 +191,9 @@ Get started in seconds with our verified environments. Click each icon below for
...
 
-|Weights and Biases|Roboflow ⭐ NEW|
-|:-:|:-:|
-|Automatically track and visualize all your YOLOv5 training runs in the cloud with [Weights & Biases](https://wandb.ai/site?utm_campaign=repo_yolo_readme)|Label and export your custom datasets directly to YOLOv5 for training with [Roboflow](https://roboflow.com/?ref=ultralytics) |
-
+|                                                                    Weights and Biases                                                                     |                                                        Roboflow ⭐ NEW                                                        |
+| :-------------------------------------------------------------------------------------------------------------------------------------------------------: | :--------------------------------------------------------------------------------------------------------------------------: |
+| Automatically track and visualize all your YOLOv5 training runs in the cloud with [Weights & Biases](https://wandb.ai/site?utm_campaign=repo_yolo_readme) | Label and export your custom datasets directly to YOLOv5 for training with [Roboflow](https://roboflow.com/?ref=ultralytics) |
```
Comment (Borda) [inline README.md]:
```suggestion
|                                                                    Weights and Biases                                                                     |                                                        Roboflow ⭐ NEW                                                        |
| :-------------------------------------------------------------------------------------------------------------------------------------------------------: | :--------------------------------------------------------------------------------------------------------------------------: |
| Automatically track and visualize all your YOLOv5 training runs | Label and export your custom datasets directly to YOLOv5 |
|  in the cloud with [Weights & Biases](https://wandb.ai/site?utm_campaign=repo_yolo_readme) |   for training with [Roboflow](https://roboflow.com/?ref=ultralytics) |
```

**Comment by Borda** [comment]
@glenn-jocher @kalenmike I have transformed the wide table to HTML divs so it is fine now :]

**Comment by Borda** [comment]
rip :confused:

**Pull request closed without merging**